prov/shm/src/smr.h

```diff
@@ -172,8 +174,8 @@ struct smr_cmd_ctx {
 
 OFI_DECLARE_FREESTACK(struct smr_rx_entry, smr_recv_fs);
 OFI_DECLARE_FREESTACK(struct smr_cmd_ctx, smr_cmd_ctx_fs);
-OFI_DECLARE_FREESTACK(struct smr_tx_entry, smr_pend_fs);
-OFI_DECLARE_FREESTACK(struct smr_sar_entry, smr_sar_fs);
+OFI_DECLARE_FREESTACK(struct smr_tx_entry, smr_tx_fs);
+OFI_DECLARE_FREESTACK(struct smr_pend_entry, smr_rx_fs);
```

Comment:
It is confusing to see both a recv_fs and rx_fs.  It's doubly confusing to see that rx_entry is associated with the recv_fs, but tx_entry with tx_fs.

Reply:
that's a very good point. can we do a bit more refactoring and take something like what zach did to merge the tx and pend entry into one progress (or pend) entry?
https://github.com/ofiwg/libfabric/pull/8281/commits/e1720109269a23bc980dd5114dba4adeed83b914
Then we can keep rx_entry -> rx_fs
and then have one type of pend_fs but two different instances of them - rx_pend_fs and tx_pend_fs?

Reply:
Do you mean you want to do this refactoring in this patch? Wouldn't it make sense to add the refactoring in a separate PR as this is not related to the target of this PR?

Reply:
I was suggesting adding a patch within this series, between the first and the async one, to merge them but I'm ok with doing that as a follow up if you prefer

Reply:
For this patch, maybe just rename rx_fs to pend_fs?

Reply:
If you guys are okay with it, I'd like to rename the rx_fs->pend_fs as Sean suggested for this PR, then I can touch base with Alexia to more fully understand the refactoring she's suggesting. I don't mind creating a separate PR for that.

Reply:
That's fine with me. Thanks!